src/test/java/org/elasticsearch/index/mapper/multifield/MultiFieldTests.java

```diff
@@ -446,4 +448,45 @@ public void testMultiFieldsInConsistentOrder() throws Exception {
             assertThat(field, equalTo(multiFieldNames[i++]));
         }
     }
+    
+    @Test
+    @SuppressWarnings("unchecked")
+    // The fielddata settings need to be in consistent order, else unneccesary mapping sync's can be triggered
+    public void testMultiFieldsFieldDataSettingsInConsistentOrder() throws Exception {
+        final String MY_MULTI_FIELD = "multi_field";
+        
+        // Possible fielddata settings
+        Map<String, Object> possibleSettings = new TreeMap<String, Object>();
+        possibleSettings.put("filter.frequency.min", 1);
+        possibleSettings.put("filter.frequency.max", 2);
+        possibleSettings.put("filter.regex.pattern", ".*");
+        possibleSettings.put("format", "fst");
+        possibleSettings.put("loading", "eager");
```

Comment:
can we add bogus non relevant settings? Would be good to know they don't mess up with things.